.github/workflows/release.yaml

```diff
@@ -166,24 +166,76 @@ jobs:
-  # publish-prql-java:
-  #   runs-on: ubuntu-latest
-  #   steps:
-  #     - name: Checkout code
-  #       uses: actions/checkout@v3
-  #     - name: Install Java and Maven
-  #       uses: actions/setup-java@v3
-  #       with:
-  #         java-version: 8
-  #     - name: Release Maven package
-  #       uses: samuelmeuli/action-maven-publish@v1
-  #       with:
-  #         gpg_private_key: ${{ secrets.gpg_private_key }}
-  #         gpg_passphrase: ${{ secrets.gpg_passphrase }}
-  #         nexus_username: ${{ secrets.nexus_username }}
-  #         nexus_password: ${{ secrets.nexus_password }}
-  #         directory: prql-java/java/
+  build-prql-lib:
+    strategy:
+      matrix:
+        include:
+          - os: macos-latest
+            directory: darwin-x86-64
+            target: x86_64-apple-darwin
+            suffix: ".dylib"
+          - os: macos-latest
+            directory: darwin-aarch64
+            target: aarch64-apple-darwin
+            suffix: ".dylib"
+          - os: ubuntu-latest
+            directory: linux-x86-64
+            target: x86_64-unknown-linux-gnu
+            suffix: ".so"
+          - os: windows-latest
+            directory: win32-x86-64
+            target: x86_64-pc-windows-msvc
+            suffix: ".dll"
+      fail-fast: false
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+      - name: Install targets
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          override: true
+          toolchain: stable
+          target: ${{ matrix.target }}
+      - name: Build
+        uses: richb-hanover/cargo@v1.1.0
+        with:
+          command: build
+          args: --target=${{ matrix.target }} --release -p prql-lib
+      - uses: actions/upload-artifact@v3
+        with:
+          name: ${{ matrix.directory }}
+          path:
+            target/${{ matrix.target }}/release/*prql_lib${{ matrix.suffix }}
+          if-no-files-found: error
+
+  publish-prql-java:
+    needs:
+      - build-prql-lib
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install JDK 8 and Maven
+        uses: actions/setup-java@v3
+        with:
+          distribution: "temurin"
+          java-version: 8
+          cache: "maven"
+      - uses: actions/download-artifact@v3
+        with:
+          path: bindings/prql-java/src/main/resources
+      - name: Release Maven package
+        uses: samuelmeuli/action-maven-publish@v1
+        with:
+          directory: bindings/prql-java
+          maven_profiles: release
+          maven_args:
+            -Drevision=0.6.1 -q --batch-mode --show-version --strict-checksums
+          gpg_private_key: ${{ secrets.GPG_PRIVATE_KEY }}
+          gpg_passphrase: ${{ secrets.GPG_PASSPHRASE }}
+          nexus_username: ${{ secrets.SONATYPE_USER }}
+          nexus_password: ${{ secrets.SONATYPE_PASSWD }}
```

Comment:
Let me know what you want me to set up here

Reply:
Yes, please. These are repository secrets required for publishing the signed artifacts to maven central. You have SONATYPE* already and I believe you have GPG key as well.

Reply:
Last time we spoke I thought you might be taking the next steps there. But I'm probably misremembering. Do you have the links to hand that I should follow to get these set up? I remember being a bit confused where to go for the Sonotype ones.

Reply:
Sure. `SONATYPE_USER` and `SONATYPE_PASSWD` are the credentials you used to login https://issues.sonatype.org/ (and https://s01.oss.sonatype.org/). https://issues.sonatype.org/browse/OSSRH-89424 is the ticket for applying Maven central access - I'll request to revoke my access there once I complete testing.

Please refer to [this](https://docs.github.com/en/authentication/managing-commit-signature-verification/generating-a-new-gpg-key) to generate GPG key if you don't have. And use `gpg --keyserver keyserver.ubuntu.com --send-keys <key id>` to publish your key(required to ensure the artifacts are signed by you, not anyone else).